### DISCUSSION TITLE
Support WASM / WASI

### DISCUSSION THREAD

**Discussion by itsbalamurali** [SeaQL/sea-orm#1348] [Ideas]
## Motivation

To be able to use Sea-Orm in WASI environments.

## Proposed Solutions

As of now, the Underlying `sqlx` is dependent on the `socket2` crate which doesn't yet support networking on wasm/wasi environments. We will have to wait for support on sqlx. But I thought its also good to have an issue/feature request to track.

## Additional Information

https://wasi.dev

**Comment by itsbalamurali** [comment]
https://github.com/tokio-rs/mio/issues/1346

**Comment by tyt2y3** [comment]
Umm interesting idea. May I know which runtime you are targeting and the system architecture?

**Comment by itsbalamurali** [comment]
Tokyo has warm support partially without the `net` feature. Target system architecture: `wasm32-wasi`

**Reply by tyt2y3** [reply]
I mean, what WASI host this can actually run on. As far as I know, wasmtime does not support async, and I am not aware of a WASI runtime that does with proper networking support.

**Comment by thangchung** [comment]
I saw WasmEdge forked `sqlx` repo, and tried to add capability for `wasi` => https://github.com/WasmEdge/sqlx

**Reply by billy1624** [reply]
Hey @itsbalamurali, thanks for the prompt! We will keep an eye on the development :P

**Reply by thangchung** [reply]
Just saw the announcement about `wasix` at https://wasmer.io/posts/announcing-wasix, I will work on the POC to see how can it works with `seaorm`. I will get back later about the result.

**Reply by billy1624** [reply]
Thanks!! Looking forward to it. Feel free to ask me for help :)

**Reply by itsbalamurali** [reply]
@thangchung @billy1624 Also, https://github.com/cloudflare/workers-rs/pull/324 & https://github.com/sfackler/rust-postgres/commit/64bf779f7c91524b820e60226a6b8c8075d2dfa4
It would be massive if we can get SeaORM working on Cloudflare workers.

**Reply by billy1624** [reply]
I'm not familiar with "Cloudflare workers". Seems it's a serverless worker?

**Reply by thangchung** [reply]
I tried with `wasix` to build the solution with `seaorm`, but have got the issue here https://github.com/wasix-org/cargo-wasix/issues/10, still waiting for the team to fix it

**Comment by ilbertt** [comment]
Any news on this? It would be really nice to see SeaORM work in WASI environments.
